sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsTestUtils.scala

```diff
@@ -198,6 +219,32 @@ trait SQLMetricsTestUtils extends SQLTestUtils {
       }
```

Comment:
we can update this in order to avoid code duplication and reuse `testSparkPlanMetrics`.

Reply:
I have updated `testSparkPlanMetrics` to invoke `testSparkPlanMetricsWithPredicates` to avoid code duplication.